qiskit/tools/apps/optimization.py

```diff
@@ -94,15 +95,19 @@ def SPSA_optimization(obj_fun, initial_theta, SPSA_parameters, max_trials,
             theta_minus_save.append(theta_minus)
             cost_plus_save.append(cost_plus)
             cost_minus_save.append(cost_minus)
+        # record circuits ran
+        circuits += circuits_plus
+        circuits += circuits_minus
 
         if k >= max_trials - last_avg:
             theta_best += theta / last_avg
 
     # final cost update
-    cost_final = obj_fun(theta_best)
+    cost_final, circuits_final = obj_fun(theta_best)
+    circuits += circuits_final
     print('Final objective function is: %.7f' % cost_final)
     return [cost_final, theta_best, cost_plus_save, cost_minus_save,
-            theta_plus_save, theta_minus_save]
+            theta_plus_save, theta_minus_save], circuits
```

Comment:
Can you update the `Returns` docstrings for this function and the others that now return the circuits?